rali/rali/source/rali_api_meta_data.cpp

```diff
@@ -141,62 +153,88 @@ RALI_API_CALL raliGetImageLabels(RaliContext p_context, int* buf)
 }
 
 unsigned
-RALI_API_CALL raliGetBoundingBoxCount(RaliContext p_context, unsigned image_idx )
+RALI_API_CALL raliGetBoundingBoxCount(RaliContext p_context, int* buf)
 {
+    unsigned size = 0;
     auto context = static_cast<Context*>(p_context);
     auto meta_data = context->master_graph->meta_data();
+    size_t meta_data_batch_size = meta_data.second->get_bb_labels_batch().size();
+    if(context->user_batch_size() != meta_data_batch_size)
+        THROW("meta data batch size is wrong " + TOSTR(meta_data_batch_size) + " != "+ TOSTR(context->user_batch_size() ))
     if(!meta_data.second)
+        THROW("No label has been loaded for this output image")
+    for(unsigned i = 0; i < meta_data_batch_size; i++)
     {
-        WRN("No label has been loaded for this output image")
-        return 0;
+        buf[i] = meta_data.second->get_bb_labels_batch()[i].size();
+        size += buf[i];
     }
-    return meta_data.second->get_bb_labels_batch()[image_idx].size();
+    return size;
 }
 
 void
-RALI_API_CALL raliGetBoundingBoxLabel(RaliContext p_context, int* buf, unsigned image_idx )
+RALI_API_CALL raliGetBoundingBoxLabel(RaliContext p_context, int* buf)
 {
     auto context = static_cast<Context*>(p_context);
     auto meta_data = context->master_graph->meta_data();
+    size_t meta_data_batch_size = meta_data.second->get_bb_labels_batch().size();
+    if(context->user_batch_size() != meta_data_batch_size)
+        THROW("meta data batch size is wrong " + TOSTR(meta_data_batch_size) + " != "+ TOSTR(context->user_batch_size() ))
     if(!meta_data.second)
     {
         WRN("No label has been loaded for this output image")
         return;
     }
-    unsigned bb_count = meta_data.second->get_bb_labels_batch()[image_idx].size();
-    memcpy(buf, meta_data.second->get_bb_labels_batch()[image_idx].data(),  sizeof(int)*bb_count);
+    auto ptr = buf;
```

Comment:
Codacy found an issue: [Variable 'ptr' is assigned a value that is never used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6302037)